### PR TITLE
Support only one endpoint

### LINKED ISSUE #29: Configuring Multiple Endpoints Triggers Indefinite Blocking Without Successful Connection
Using more than one endpoint in the configuration leads to an indefinite blocking without a successful connection.

The custom build, created with blocking deactivated, exposes the root cause of the issue:
> {"level":"error","timestamp":"2024-02-26T11:24:03Z","logger":"setup","caller":"workspace/main.go:226","msg":"unable to connect to duros","error":"rpc error: code = Unavailable desc = last connection error: connection error: desc = \"transport: Error while dialing: dial tcp: address 10.138.1.11:443,10.138.1.21:443,10.138.1.31:443: too many colons in address\"","stacktrace":"main.main\n\t/workspace/main.go:226\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}

The defect was introduced with issue #11.

### PR DESCRIPTION
**Pull request by robertvolkmann**
To support multiple endpoints use a DNS record that returns multiple IP addresses

Fix #29

### CODE REVIEW

**Comment by majst01** [comment]
TODO:
- [x] try with cloud-api, api-server

File: client.go [hunk 1]
```diff
@@ -318,3 +308,25 @@ func (c ByteCredentials) getTransportCredentials() (credentials.TransportCredent
...
 }
+
+func isValid(endpoint string) error {
+	host, port, err := net.SplitHostPort(endpoint)
+	if err != nil {
+		//nolint dunno howto convert this to errors.As
+		if addrErr, ok := err.(*net.AddrError); ok {
+			return fmt.Errorf("%s", addrErr.Err)
+		}
+		// shouldn't happen, but...
```
Comment (majst01) [inline client.go]: I think this extra error handling can be skipped

File: client.go [hunk 2]
```diff
@@ -318,3 +308,25 @@ func (c ByteCredentials) getTransportCredentials() (credentials.TransportCredent
...
+		return err
```
Comment (robertvolkmann) [inline client.go]:
```suggestion
		return err
```

File: client.go (hunk 1 above)
Comment (robertvolkmann) [reply client.go]: Done

**Review by majst01 — APPROVED** [review]

**Pull request merged**